### PR TITLE
refactor: change polymarket provider to use markets endpoint

### PR DESCRIPTION
**Pull request by technicallyty**
- polymarket provider now uses the `markets` endpoint.
- offchain ticker is now expected to be <market_id>/<token_id>
- removed 1.00 normalization per dydx request

### CODE REVIEW

File: .golangci.yml [hunk 1]
```diff
@@ -16,7 +16,6 @@ linters:
     - dogsled
     # Check for two durations multiplied together.
     - durationcheck
-    - exportloopref
```
Comment (technicallyty) [inline .golangci.yml:19]: removing this linter because its no longer applicable in go1.22+ https://go.dev/blog/loopvar-preview

**Review by wesl-ee — APPROVED** [review]

File: providers/apis/polymarket/api_handler.go [hunk 2]
```diff
@@ -21,24 +17,14 @@ const (
 	// Name is the name of the Polymarket provider.
 	Name = "polymarket_api"
 
-	host = "clob.polymarket.com"
 	// URL is the default base URL of the Polymarket CLOB API. It uses the midpoint endpoint with a given token ID.
-	URL = "https://clob.polymarket.com/midpoint?token_id=%s"
+	URL = "https://clob.polymarket.com/markets/%s"
 
 	// priceAdjustmentMax is the value the price gets set to in the event of price == 1.00.
```
Comment (wesl-ee) [inline providers/apis/polymarket/api_handler.go]: Can remove this comment since priceAdjustmentMax not relevant anymore

Comment (technicallyty) [reply providers/apis/polymarket/api_handler.go]: updated [fix comment](https://github.com/skip-mev/slinky/pull/666/commits/3700ff25371334bc037c6e93ae8e4864bdf5ff87)

**Review by aljo242 — APPROVED** [review]

**Comment by aljo242** [comment]
LGTM.  Just want to make sure the deployment is stable before we rip the new release

**Comment by technicallyty** [comment]
tested this locally. prices work. tested a resolved market, got 1 for a resolved win, and 0 for a resolved loss. got the expected values for the other live markets. merging this.

**Pull request merged**